src/Diagnostics.RuntimeHost/Middleware/DiagnosticsRequestMiddleware.cs

```diff
@@ -110,7 +110,13 @@ private void BeginRequest_Handle(HttpContext httpContext)
                 clientPrincipalName = httpContext.Request.Headers[ClientPrincipalNameHeader];
             }
 
-            httpContext.Items.Add(HostConstants.DataProviderContextKey, new DataProviderContext(dataSourcesConfigurationService.Config, values.FirstOrDefault() ?? string.Empty, cTokenSource.Token, startTimeUtc, endTimeUtc, wawsObserverTokenService, supportBayApiObserverTokenService, clientObjId, clientPrincipalName, kustoHeartBeatService));
+            string geomasterName = string.Empty;
+            if (httpContext.Request.Headers.ContainsKey(GeomasterNameHeader))
+            {
```

Comment:
Will this header be sent for all requests regional and global? We can do `else geoMaster=<global-geo>` for safety

Reply:
That default logic is in the GeoMasterCertClient constructor:
  var geoEndpoint = new UriBuilder(geoMasterName ?? configuration.GeoEndpointAddress)
But I noticed that geoMasterName is Empty if not supplied, so I will update the logic.

Reply:
Updated